components/butterfly/src/server/mod.rs

```diff
@@ -1231,6 +1252,96 @@ impl<'a> Serialize for ServerProxy<'a> {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use habitat_core::service::ServiceGroup;
+
+    impl Default for Service {
+        fn default() -> Self {
+            Service {
+                member_id: String::default(),
+                service_group: ServiceGroup::new(None, "service", "group", None).unwrap(),
+                incarnation: Default::default(),
+                initialized: Default::default(),
+                pkg: Default::default(),
+                cfg: Default::default(),
+                sys: Default::default(),
+            }
+        }
+    }
+
+    #[test]
+    fn elections_are_restarted_when_leader_health_is_unknown() {
```

Comment:
Would it be possible / practical to extract the setup for this test and the following one, in order to clarify exactly which parts are changing between them, and to also perhaps facilitate the addition of tests around other leader health states?

Reply:
Yes, I just hadn't gotten there yet. I'll plan to do this is a follow-up PR where I flesh out the testing more. I wanted to get this one out to fix the bug ASAP. I think when I add more tests for `elections_to_restart_impl` the proper refactoring will become more apparent.